src/stack.cpp

```diff
@@ -197,7 +191,7 @@ static struct HSTreeInterface stack_nth_child(HSTree root, size_t idx) {
     return intface;
 }
 
-static size_t stack_child_count(HSTree root) {
+static size_t stack_child_count(std::shared_ptr<HSStack> root) {
```

Comment:
What's the deal with this method? Should we get rid of it?

Reply:
I'm not sure if it was *this* method, but there were some that appeared entirely unused. As I said in my global comment, please advise what can be simply removed.

Reply:
If `stack_child_count` is not explicitely used, we can remove it.

Reply:
From what I can tell: It's called in `slice_child_count`, which in turn is referenced only by `layer_nth_child`, which in turn is referenced only by `stack_nth_child`, which in turn is referenced only by `slice_nth_child`.

So I wouldn't claim to understand any of that, but it sure looks like a bowl of unused spaghetti to me.

Reply:
I think the `slice_nth_child` is used or will be used for the tree interface (that's the interface that is passed to the tree printing algorithm, e.g. in the `herbstclient stack` command)